app/components/meeting/new-meeting-modal.js

```diff
@@ -0,0 +1,226 @@
+  }
+
+  get isAnnexMeeting() {
+    return isAnnexMeetingKind(this.selectedKindUri);
+  }
+
+  async generateNewNumber() {
+    const meetings = await this.store.query('meeting', {
+      filter: {
+        ':gte:planned-start': new Date(this.currentYear, 0, 1).toISOString(),
+      },
+      sort: '-number',
+    });
+
+    if (meetings.length) {
+      const meetingsFromThisYear = meetings.filter(
+        (meeting) =>
+          meeting.plannedStart &&
+          meeting.plannedStart.getFullYear() === this.currentYear
+      );
+      const meetingIds = meetingsFromThisYear
+        .map((meeting) => meeting.number)
+        .filter((meetingId) => meetingId !== undefined);
+      let id = 0;
+      // FIX voor de eerste agenda van het jaar -> Anders math.max infinity
+      if (meetingIds.length !== 0) {
+        id = Math.max(...meetingIds);
+      }
+      this.meetingNumber = id + 1;
+      this.formattedMeetingIdentifier = `VR PV ${this.currentYear}/${this.meetingNumber}`;
+    }
+  }
+
+  async createAgenda(meeting, date) {
+    const status = await this.store.findRecordByUri(
+      'agendastatus',
+      CONSTANTS.AGENDA_STATUSSES.DESIGN
+    );
+    const fallBackDate = this.formatter.formatDate(null);
+    const agenda = this.store.createRecord('agenda', {
+      serialnumber: 'A',
+      title: `Agenda A voor zitting ${moment(meeting.plannedStart).format(
+        'D-M-YYYY'
+      )}`,
+      createdFor: meeting,
+      status,
+      created: date || fallBackDate,
+      modified: date || fallBackDate,
+    });
+    const savedAgenda = await agenda.save();
+    return savedAgenda;
+  }
+
+  // new meeting parameter prevents extra request of agenda.createdFor
+  async createAgendaitemToApproveMinutes(agenda, newMeeting, closestMeeting) {
+    const now = new Date();
+
+    // load code-list item
+    const decisionResultCode = await this.store.findRecordByUri(
+      'decision-result-code',
+      CONSTANTS.DECISION_RESULT_CODE_URIS.GOEDGEKEURD
+    );
+
+    // Treatment of agenda-item / decision activity
+    const startDate = newMeeting.plannedStart;
+    const agendaItemTreatment = this.store.createRecord(
+      'agenda-item-treatment',
+      {
+        created: now,
+        modified: now,
+        startDate: startDate,
+        decisionResultCode,
+      }
+    );
+    await agendaItemTreatment.save();
+
+    const agendaitem = this.store.createRecord('agendaitem', {
+      created: now,
+      agenda,
+      number: 1,
+      shortTitle: `Goedkeuring van het verslag van de vergadering van ${moment(
+        closestMeeting.plannedstart
+      ).format('dddd DD-MM-YYYY')}.`,
+      formallyOk: CONSTANTS.ACCEPTANCE_STATUSSES.NOT_YET_OK,
+      mandatees: [],
+      pieces: [],
+      approvals: [],
+      isApproval: true,
+      treatments: A([agendaItemTreatment]),
+    });
+    return await agendaitem.save();
+  }
+
+  @task({ drop: true })
+  *createNewSession() {
+    const {
+      isDigital,
```

Comment:
Doesn't seem to be used and is not defined on the `meeting` model. Can be removed.